### ISSUE TITLE
not working on amazon ec2 instance remotely

### ISSUE THREAD

**Issue by neocxf** [wurstmeister/kafka-docker#183]
I use the image on Amazon AMI ec2 instance.  I modified the docker-compose.yml as follows:
```
  zookeeper:
    image: wurstmeister/zookeeper
    ports:
      - "2181:2181"
  kafka:
    build: .
    ports:
        - "9092:9092"
    links:
        - zookeeper:zookeeper
    environment:
      KAFKA_ADVERTISED_HOST_NAME: <my ec2 internal ip address>
      #HOSTNAME_COMMAND: "wget -t3 -T2 -qO-  http://169.254.169.254/latest/meta-data/local-ipv4"
      KAFKA_ZOOKEEPER_CONNECT: zookeeper:2181
      KAFKA_ADVERTISED_PORT: 9092
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
```

Right now, I can use the official's recommended [tutorial](http://wurstmeister.github.io/kafka-docker/) to successfully finish the producer and consumer script verification on the Amazon EC2 instance.

But, I want to access my kafka service **remotely.** And when I try to access the kafka service using the public IP address of ec2 instance, I get the following error
```
error { Error: connect ETIMEDOUT 172.21.1.13:9092
    at Object.exports._errnoException (util.js:1018:11)
    at exports._exceptionWithHostPort (util.js:1041:20)
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1090:14)
  code: 'ETIMEDOUT',
  errno: 'ETIMEDOUT',
  syscall: 'connect',
  address: '172.21.1.13',
  port: 9092 }

```

As you can see, the consumer client says that the ip is unreachable, since the ip is just the internal ip address of Amazon ami services.

So I try to change the KAFKA_ADVERTISED_HOST_NAME with the public ip of ec2, another kinds of problems occur.  I have already dig through the internet already, but didn't find a solution to really solve the problem.

Could your guys kindly help me?

**Comment by smifun** [comment]
I run into similar issue when I tried to run Kafka on Kubernetes on AWS. My problem was already described here: 
http://stackoverflow.com/questions/41868161/kafka-in-kubernetes-cluster-how-to-publish-consume-messages-from-outside-of-kub.

**Comment by sscaling** [comment]
@neocxf - is this resolved after PR #199 ?

**Comment by sscaling** [comment]
Closing due to staleness. 

If this is still an issue, please make sure you understand the concepts in the wiki page: https://github.com/wurstmeister/kafka-docker/wiki/Connectivity
